android/app/src/main/java/updated/mysterium/vpn/ui/settings/SettingsActivity.kt

```diff
@@ -176,4 +207,30 @@ class SettingsActivity : BaseActivity() {
         ).toInt()
         listPopupWindow.height = popUpHeight - margin // add margin from bottom
     }
+
+    private fun showNatCompatibilityPopUpWindow() {
+        val inflater = getSystemService(Context.LAYOUT_INFLATER_SERVICE) as LayoutInflater
+        val bindingPopUpView = ViewItemNatCompatibilityDescriptionBinding.inflate(inflater)
+
+        natCompatibilityPopUpWindow = PopupWindow(
+            bindingPopUpView.root,
+            POPUP_WINDOW_WIDTH_DP.px,
+            LinearLayout.LayoutParams.WRAP_CONTENT
+        )
+
+        TransitionManager.beginDelayedTransition(bindingPopUpView.root)
+        natCompatibilityPopUpWindow?.showAtLocation(
+            binding.root,
+            Gravity.TOP,
+            POPUP_WINDOW_END_OFFSET_DP.px,
+            getPopUpWindowVerticalOffset()
+        )
+    }
+
+    private fun getPopUpWindowVerticalOffset(): Int {
+        val baseViewRectangle = binding.root.calculateRectOnScreen()
+        val hintButtonViewRectangle = binding.natHelperFrameButton.calculateRectOnScreen()
+        val distance = abs(baseViewRectangle.top - hintButtonViewRectangle.bottom).toInt()
+        return distance + POPUP_WINDOW_TOP_OFFSET_DP.px
+    }
```

Comment:
You need to incapsulate this dialog in separate class. So you can easily reuse it.